Player to userid

yuriyoleynikov opened the pull request.

yuriyoleynikov committed:
-Add ChangePlayersEvent class in events
-Change StartEvent class, add filds
-Change On(StartEvent gameEvent) method
-Add On(ChangePlayersEvent gameEvent) method
-Change MatchDetailsViewModel class, add fild
-Change MatchModel class, add filds

yuriyoleynikov committed:
-Change Details view on MatchController class
-Add PlayerToUser action in MatchController class
-Add UserStats action in StatsController class

yuriyoleynikov committed:
-Add GetMatchesWithUser method in DatabaseMatchListService class
-Add PlayerToUserAsync method in DatabaseMatchService class

yuriyoleynikov committed:
-Change DynamicStatsService class!
-Add GetMatchesWithUser method in IMatchListService
-Add PlayerToUserAsync method in IMatchService
-Add GetMatchesWithUser in InMemoryMatchListService
-Add PlayerToUserAsyncin InMemoryMatchService
-Add GetUserStatsModel method in IStatsService

yuriyoleynikov committed:
-Add Player to User functionality in Details view
-Add UserStats link in _Layout file
-Add UserStats view
-To comment HandlerSnapshotTests class

Igorbek commented on src/YoTennis/Controllers/MatchController.cs:
```diff
@@ -48,16 +49,54 @@ public MatchController(UserManager<ApplicationUser> userManager, IMatchListServi
+            };
+
+            if (player == Player.First)
+                e.FirstPlayerUserId = "none";
```
use `null`

yuriyoleynikov replied:
ok

Igorbek commented on src/YoTennis/Models/MatchDetailsViewModel.cs:
```diff
@@ -9,5 +9,6 @@ public class MatchDetailsViewModel
     {
         public string Id { get; set; }
         public MatchModel MatchModel { get; set; }
+        public string UserId { get; set; }
```
do not expose user id, just use user name

Igorbek commented on src/YoTennis/Services/DatabaseMatchService.cs:
```diff
@@ -120,5 +120,12 @@ public Task Reset()
             return _gameHandler.PlayersStats;
         }
+
+        public async Task PlayerToUserAsync(Player player)
```
what does this method do?

Igorbek commented on tests/YoTennis.Tests/Test/HandlerSnapshotTests.cs:
```diff
@@ -11,7 +11,7 @@ namespace YoTennis.Tests.Test
 
     public class HandlerSnapshotTests
-    {
+    {/*
```
why did you comment the tests?

Igorbek commented:
Are you going to complete/merge this?

Commit:
Merge master

Commit:
Resolve merge

Commit:
-Add DeletePlayersEvent class.
-Change method PlayerToUserDelete in MatchController class.
-Change GameHandler class.

Commit:
-Remove PlayerToUserAsync method in DatabaseMatchService class, InMemoryMatchService class, IMatchService
-Change tests in HandlerSnapshotTests
